### PR TITLE
[@entropy/winston-elasticsearch-apm] feat: adding types to @entropy/winston-elasticsearch-apm

### PR DESCRIPTION
**Pull request by mykvas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by michael-vasyliv** [comment]
**note** there was deleted `elastic-apm-node` from `notNeededPackages.json` because of this PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34243

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## entropy__winston-elasticsearch-apm ([<kbd>unpkg</kbd>](https://unpkg.com/browse/entropy__winston-elasticsearch-apm@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'entropy__winston-elasticsearch-apm'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. The declaration doesn't match the JavaScript module 'entropy__winston-elasticsearch-apm'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7623b9dd718b90583d3c118dc6936a4e46bf8fee
</p>

File: types/entropy__winston-elasticsearch-apm/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    apm: typeof Agent;
+}
+
+declare class ElasticsearchApm extends TransportStream {
+    constructor(opts: ElasticsearchApmOptions);
+
+    log(info: any, next: () => void): any;
+}
+
+export default ElasticsearchApm;
```
Comment (weswigham) [inline types/entropy__winston-elasticsearch-apm/index.d.ts:19]: The `cjs` usage of this package in the `README` is

`var ElasticsearchApm = require('@entropy/winston-elasticsearch-apm');` - please use an `export =` here to support that, and don't disable the lint rules requesting you use an `export =`, and then turn on `esModuleInterop` in the `tsconfig` if you'd still like to use `default` imports in the tests.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/entropy__winston-elasticsearch-apm/index.d.ts (hunk 1 above)
Comment (michael-vasyliv) [reply types/entropy__winston-elasticsearch-apm/index.d.ts:19]: @weswigham done

File: types/entropy__winston-elasticsearch-apm/tslint.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "npm-naming": [
```
Comment (weswigham) [inline types/entropy__winston-elasticsearch-apm/tslint.json:4]: Hopefully you shouldn't need this lint exception at all anymore? Right?

Comment (michael-vasyliv) [reply types/entropy__winston-elasticsearch-apm/tslint.json]: @weswigham yes, you are right. It's fixed.

File: types/entropy__winston-elasticsearch-apm/tslint.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {}
```
Comment (weswigham) [inline types/entropy__winston-elasticsearch-apm/tslint.json:3]: you should be able to remove this field entirely 😉

Comment (michael-vasyliv) [reply types/entropy__winston-elasticsearch-apm/tslint.json]: @weswigham done

**Review by weswigham — APPROVED** [review]

**Comment by michael-vasyliv** [comment]
Ready to merge

**Pull request merged**